sqlair.go

```diff
@@ -372,7 +524,20 @@ func (tx *TX) Query(ctx context.Context, s *Statement, inputArgs ...any) *Query
 	if ctx == nil {
 		ctx = context.Background()
 	}
+	if tx.isDone() {
+		return &Query{ctx: ctx, err: ErrTXDone}
+	}
+
+	sqlstmt, err := tx.db.prepareStmt(ctx, tx.sqlconn, s)
+	if err != nil {
+		return &Query{ctx: ctx, err: err}
+	}
 
 	qe, err := s.pe.Query(inputArgs...)
-	return &Query{qs: tx.tx, qe: qe, ctx: ctx, err: err}
+	if err != nil {
+		return &Query{ctx: ctx, err: err}
+	}
+
+	sqlstmt = tx.sqltx.Stmt(sqlstmt)
```

Comment:
Instead of doing this here, let's do it at execution time so that we can close the stmt one immediately.

Reply:
I've left this as is, see comment above about the stmt.